### DISCUSSION TITLE
Add support to publish to geometry_msgs/TwistStamped using the teleop panel

### DISCUSSION THREAD

**Discussion by TFM99102** [foxglove/community#733] [Q&A]
So far the  teleop panel can only publish an `geometry_msgs/Twist`  message. I think it would be useful to also be able to publish a `geometry_msgs/TwistStamped` message.

**Comment by jtbandes** [comment]
Internal tracking issue: https://linear.app/foxglove/issue/FG-3786

**Reply by bernard-bear** [reply]
Hi @jtbandes, can I check if there's any update on whether this feature is being worked on and if there's any timeline? If not, I'm happy to attempt a PR (any guidance would be appreciated) as I need this feature as well. Thanks!

**Reply by jtbandes** [reply]
We aren't planning to prioritize this in the near future, but if you are interested in making a PR we can review it. The Teleop code is here: https://github.com/foxglove/studio/blob/main/packages/studio-base/src/panels/Teleop/TeleopPanel.tsx

One thing I would wonder about (not sure what the right answer is) is what timestamp should be used? Is it meant to be the system clock or the latest time from the connected data source, or what? I imagine there will also need to be a way to choose the frame id in the settings.

**Reply by bernard-bear** [reply]
For a start, I would like to publish `TwistStamped` without any header. Would that be sufficient as as standalone PR? (For my purposes, I do not need the information in the header)

**Reply by jtbandes** [reply]
It seems strange to have a TwistStamped without filling in the header values – in that case that's what the Twist message is designed for. What is the use case this is solving?

**Comment by tonynajjar** [comment]
[Nav2 migrated to TwistStamped](https://discourse.ros.org/t/notice-nav2-migrated-to-twiststamped-to-replace-twist-for-cmd-vel-topics/40944), this is becoming more urgent for the teleop panel to stay relevant. @jtbandes any update on that internal ticket?

**Reply by defunctzombie** [reply]
There are no updates to that issue. For folks that must have this functionality asap (and are blocked) you can make a custom panel to produce any kind of message suited for your robot.

**Reply by jtbandes** [reply]
Interesting to note that they have also added a `enable_stamped_cmd_vel` parameter for toggling between types. It's possible we could autodetect this. However, I believe in ROS 2 parameters are per-node so I'm not sure this would actually work
